### DISCUSSION TITLE
Facing issues while using liquid expression, any documentation?

### DISCUSSION THREAD

**Discussion by SachinMehndiratta** [elsa-workflows/elsa-core#1101] [Q&A]
Code snippets for reference while triggering signaledworkflow:

  var test = new Test
            {
                ServerName ="ftp://testserver.com",
                PortNumber = 22,
             
            };

        //    var input = new Variables();

           //    input.Set("Test", test);

            await _signaler.TriggerSignalAsync("TriggerProcess", test)

On the subsequent dashboard activity UI;

setting portnumber as :   {{input.PortNumber}} 

Have tried different combinations but no success.

Any help?

**Comment by sfmskywalker** [comment]
Maybe try: input.Test.PortNumber.

**Reply by sfmskywalker** [reply]
And I’m not sure but Liquid may be case sensitive, so it would have to be: Input.Test.PortNumber

**Reply by SachinMehndiratta** [reply]
@sfmskywalker 
Have tried but no luck

Here are the code snippets

var test = new Test
{
ServerName ="ftp://testserver.com",
PortNumber = 22,
};
 await _signaler.TriggerSignalAsync("TriggerProcess", test)

On the subsequent dashboard activity UI;

have tried setting portnumber as : {{input.PortNumber}} , {{Input.Test.PortNumber}} , {{Input.test.PortNumber}} , {{test.PortNumber}}


Properties in the Activity are defined like

  [ActivityProperty(Hint = "Enter an expression that evaluates to the name of the host to connect.", SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })]
 public string ServerName { get; set; } = default!;

**Reply by sfmskywalker** [reply]
I see, I got confused by your initial code snippet where I thought you were using Variables, but that is commented out.

When using liquid, you need to ensure that any types you want to access are whitelisted, like so:

```csharp
public class ConfigureLiquidEngine : INotificationHandler<EvaluatingLiquidExpression>
{
   public Task Handle(EvaluatingLiquidExpression notification, CancellationToken cancellationToken)
  {
      notification.TemplateContext.Options.MemberAccessStrategy.Register<Test>();
  }
}
```

Make sure to register the notification handler with DI, like so:

```csharp
services.AddNotificationHandler<ConfigureLiquidEngine>()
```

I will update the documentation to mention this.

**Reply by SachinMehndiratta** [reply]
@sfmskywalker 
I have already implemented this all, there is a bug but I got this working with

{{Input.PortNumber}}

Bug is that published changes to workflow via UI are reflected after server restart, I perhaps raised this issue earlier as well.  I restarted the server and everything works smoothly.

Thanks for looking into this and we have a bug to handle now.

**Reply by sfmskywalker** [reply]
Good. I don't recall seeing that issue though, could you point me to it please? I want to make sure that's fixed for the 2.1 release.

**Reply by SachinMehndiratta** [reply]
It was discussed somewhere in Q&A.
Nevermind, I have created a ticket for this

https://github.com/elsa-workflows/elsa-core/issues/1107
